app/src/parsing/parsing/py_graph.py

```
@@ -0,0 +1,478 @@
+                    )
+
+    print(f"DEBUG: references count={len(references)}, defines count={len(defines)}")
+    if references:
+        print(f"DEBUG: sample ref idents: {list(references.keys())[:5]}")
+    if defines:
+        print(f"DEBUG: sample define idents: {list(defines.keys())[:5]}")
+
+    for ident, refs in references.items():
+        target_nodes = defines.get(ident, set())
+
+        for source, line, end_line, src_class, src_method in refs:
```

Ruff (0.15.10)

[warning] Loop control variable `src_class` not used within loop body

Rename unused `src_class` to `_src_class`

(B007)